src/NFun.Fuspec.TestHandler/FuspecTestBase.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using FuspecTests;
+using Nfun.Fuspec.Parser.Model;
+using NFun.Fuspec.TestHandler.Models;
+using NFun.ParseErrors;
+using NFun.Runtime;
+using NUnit.Framework.Internal;
+
+namespace NFun.Fuspec.TestHandler
+{
+    public class FuspecTestBase
+    {
+        private readonly FuspecTestCase[] _fuspecTestCases;
+
+        public FuspecTestBase(FuspecTestCase[] fuspecTestCases)
+        {
+            _fuspecTestCases = fuspecTestCases 
+                              ?? throw new ArgumentNullException("No fuspec test cases");
+        }
+
+        public IEnumerable<BaseFuspecTestCaseResult> RunTests()
+        {
+            var result = new List<BaseFuspecTestCaseResult>();
+            result.AddRange(_fuspecTestCases.Select(FuspecTestCasesRunner.RunFuspecTest));
+            return result;
+        }
+
+    }
+
+    public static class FuspecTestCasesRunner
+    {
+        public static BaseFuspecTestCaseResult RunFuspecTest(FuspecTestCase testCase)
+        {
+            var stopwatch = Stopwatch.StartNew();
+            BaseFuspecTestCaseResult testResult;
+            FunRuntime result;
+
+            try
+            {
+                result = FunBuilder.With(testCase.Script).Build();
+                testResult = GoodTestResult(stopwatch);
+            }
+            catch (FunParseException parseException)
+            {
+                testResult = BadTestResult();
+            }
+            catch (FunRuntimeException runtimeException)
+            {
+                testResult = BadTestResult();
```

Comment:
How do you want to pass fun-runtime-exception details to bad result? 

Fun runtime exception mean user script bug (that is ok) 
While other exceptions mean nfun bug (that need to be repaired)

It seems you need two bad result types

Reply:
It turns out that it is necessary to show only two types of errors - FunRuntimeException and Exception, already there is ExceptionTestCaseResult, which incorporates all other errors, is this enough or should I add another one?

Reply:
At the step of build - you can catch two type of exceptions:
- FunParseException, witch contains errorCode:Enum and additionalErrorInformation:Text
- Other exceptions -  indicator that something totally wrong and there is bug  inside nfun parser